src/shared/mount-util.c

```diff
@@ -77,13 +77,25 @@ int umount_recursive(const char *prefix, int flags) {
         return n;
 }
 
-static int get_mount_flags(const char *path, unsigned long *flags) {
-        struct statvfs buf;
+/* Get the mount flags for the mountpoint at "path" from "table" */
+static int get_mount_flags(const char *path, unsigned long *flags, struct libmnt_table *table) {
+        struct libmnt_fs *fs = NULL;
+        const char *opts = NULL;
+        int r = 0;
+
+        fs = mnt_table_find_target(table, path, MNT_ITER_FORWARD);
+        if (!fs) {
+            r = -ENOENT;
+        } else {
+            opts = mnt_fs_get_vfs_options(fs);
+            r = mnt_optstr_get_flags(opts, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+            /* relatime is default and trying to set it in an unprivileged container causes EPERM */
+            *flags &= ~MS_RELATIME;
```

Comment:
as a matter of coding style, please do not clobber output parameters of functions on failure. we generally follow the rule that our functions should initialize all output parameters on success and leave them untouched on failure. Please follow this here too. here, if mnt_optstr_get_flags() fails you still manipulate *flags, which is an output parameter. Please don't.

Also see: https://systemd.io/CODING_STYLE#code-organization-and-semantics